### PR TITLE
Handle ShuffleManager api calls when plugin is not fully initialized

### PR DESCRIPTION
**Pull request by abellina**
Signed-off-by: Alessandro Bellina <abellina@nvidia.com>

Fixes: https://github.com/NVIDIA/spark-rapids/issues/1513

Marking this as a draft as I need to run a couple of more tests.

### CODE REVIEW

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala [hunk 1]
```diff
@@ -232,10 +231,21 @@ abstract class RapidsShuffleInternalManagerBase(conf: SparkConf, isDriver: Boole
...
 
+  // Code that expects the shuffle catalog to be initialized gets it this way,
+  // with error checking in case we are in a bad state.
+  private def getCatalogOrThrow: ShuffleBufferCatalog = {
+    val catalog = GpuShuffleEnv.getCatalog
+    if (catalog == null) {
+      throw new IllegalStateException("The ShuffleBufferCatalog is not initialized but the " +
+        "RapidsShuffleManager is configured")
+    }
+    catalog
```
Comment (gerashegalov) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala:242]: nit: we can abbreviate this code with this kind of idiom:
```
Option(GpuShuffleEnv.getCatalog).getOrElse(
  throw new IllegalStateException("The ShuffleBufferCatalog is not initialized but the " +
        "RapidsShuffleManager is configured"
)
```

**Review by jlowe — APPROVED** [review]

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala [hunk 2]
```diff
@@ -256,7 +267,8 @@ abstract class RapidsShuffleInternalManagerBase(conf: SparkConf, isDriver: Boole
 
         override def getShuffleBufferMetas(sbbId: ShuffleBlockBatchId): Seq[TableMeta] = {
           (sbbId.startReduceId to sbbId.endReduceId).flatMap(rid => {
-            catalog.blockIdToMetas(ShuffleBlockId(sbbId.shuffleId, sbbId.mapId, rid))
+            catalog.blockIdToMetas(
+              ShuffleBlockId(sbbId.shuffleId, sbbId.mapId, rid))
```
Comment (tgravescs) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala:271]: nit: this looks unnecessary

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala (hunk 1 above)
Comment (abellina) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala:242]: I'll get this changed shortly.

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala (hunk 2 above)
Comment (abellina) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala:271]: Will fix.

**Comment by abellina** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala [hunk 3]
```diff
@@ -215,7 +215,6 @@ abstract class RapidsShuffleInternalManagerBase(conf: SparkConf, isDriver: Boole
   // so they all need to be lazy values that are executed when things are first called
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala:215]: Need 2021 copyrights added to this file.  Sorry, missed this on initial review.

**Review by jlowe — CHANGES_REQUESTED** [review]

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala (hunk 3 above)
Comment (abellina) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala:215]: I completely forgot. Updating.

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Pull request merged**